sklearn/gaussian_process/gpr.py

```diff
@@ -132,6 +132,21 @@ def optimizer(obj_func, initial_theta, bounds):
     log_marginal_likelihood_value_ : float
         The log-marginal-likelihood of ``self.kernel_.theta``
 
+    Examples
+    --------
+    >>> from sklearn.datasets import make_friedman2
+    >>> from sklearn.gaussian_process import GaussianProcessRegressor
+    >>> from sklearn.gaussian_process.kernels import DotProduct, WhiteKernel
+    >>> X, y = make_friedman2(n_samples=500, noise=0, random_state=0)
+    >>> #kernel = DotProduct(sigma_0_bounds=(1e-3, 1e3))
+    >>> kernel = DotProduct() + WhiteKernel()
+    >>> gpr = GaussianProcessRegressor(kernel=kernel,
+    ...         random_state=0).fit(X, y)
+    >>> gpr.score(X, y) # doctest: +ELLIPSIS
+    0.3680...
+    >>> gpr.predict(X[:2,:], return_std=True) # doctest: +ELLIPSIS
+    (array([653.0..., 592.1...]), array([316.6..., 316.6...]))
```

Comment:
That's a big uncertainty!

Reply:
Yeah I know. But it's already pretty tricky to come up with an example which is short, and doesn't give any convergence warning or something. GPs are just pretty unstable, and all example I've seen which actually work, are with over engineering of the prior and on low dimensional data (and I had to work with them for a while as a part of my thesis).

I can take one of those one dimensional examples, generate data, and have it here. It'll be a bit more code, but it'll be clean results. Should I?

Reply:
I think this is okay. I suppose we could illustrate its ability to fit a straight line with gaussian noise and to recover the variance; but then we're ignoring the interesting part being the kernels.